app/src/main/java/com/example/dragonx/NetworkService/NetworkService.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.dragonx.NetworkService
+
+import com.example.dragonx.util.Constants.Companion.BASE_URL
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class NetworkService {
+    companion object {
+        fun getInstance(): NetworkService {
+            val instance = NetworkService()
+            return instance
```

Comment:
Every call of function creates a new class instance. If this function aimed to provide the singleton it doesn't work.